crates/sui-indexer/migrations/pg/2023-08-19-044023_objects/up.sql

```diff
@@ -102,3 +102,16 @@ CREATE INDEX objects_snapshot_owner ON objects_snapshot (owner_type, owner_id, o
 CREATE INDEX objects_snapshot_coin ON objects_snapshot (owner_id, coin_type, object_id) WHERE coin_type IS NOT NULL AND owner_type = 1;
 CREATE INDEX objects_snapshot_package_module_name_full_type ON objects_snapshot (object_type_package, object_type_module, object_type_name, object_type);
 CREATE INDEX objects_snapshot_owner_package_module_name_full_type ON objects_snapshot (owner_id, object_type_package, object_type_module, object_type_name, object_type);
+
+-- versions table to store the version history of objects,
+-- input version can be NULL for created and unwrapped objects,
+-- output version is always NOT NULL. even for deleted and wrapped objects.
+CREATE TABLE objects_version (
+    object_id                   bytea         NOT NULL,
+    input_version               bigint,
+    output_version              bigint        NOT NULL,
+    checkpoint_sequence_number  bigint        NOT NULL,
+    CONSTRAINT objects_version_pk PRIMARY KEY (checkpoint_sequence_number, object_id, output_version)
```

Comment:
one thing im uncertain about is whether we want `checkpoint_sequence_number` as the first of the primary key index? I might be misguided but perhaps the primary key can just be `(object_id, output_version)` and then we can have an index on `(checkpoint_sequence_number ...)`?

Reply:
or perhaps we dont need a primary key to begin with?

Reply:
good point, I was thinking of using primary key for the cp query too, but as we will have a separate index anyways, the primary key can be indeed only object_id, object_version.

Reply:
it's possible to have a table without primary key in PG per se, but diesel requires it 
https://github.com/diesel-rs/diesel/issues/1661